src/oneid/jwts.py

```
@@ -0,0 +1,373 @@
+    header = None
+    try:
+        header = json.loads(header_json)
+        logger.debug('parsed header, header=%s', header)
+    except ValueError:
+        logger.debug('invalid header, not valid json: %s', header_json)
+        return False
+    except Exception:  # pragma: no cover
+        logger.debug('unknown error verifying header: %s', header, exc_info=True)
+        return False
+
+    if strict_jwt:
+        keys = {k: 1 for k in header}
+        for key, value in MINIMAL_JWT_HEADER.items():
+            if key not in header or header.get(key, None) != value:
+                logger.debug('invalid header, missing or incorrect %s: %s', key, header)
+                return False
+            keys.pop(key, None)
+        keys.pop('kid', None)
+
+        if len(keys) > 0:
+            logger.debug('invalid header, extra elements: %s', header)
+            return False
+    else:
+        if 'typ' not in header or header['typ'] not in ['JWT', 'JOSE', 'JOSE+JSON']:
+            logger.debug('invalid "typ" in header: %s', header)
+            return False
+
+        if 'alg' not in header or header['alg'] != 'ES256':
+            logger.debug('invalid "alg" in header: %s', header)
+            return False
+
+    logger.debug('returning %s', header)
+    return header
+
+
+def _verify_claims(payload):
+    try:
+        claims = json.loads(payload)
+        now = int(time.time())
+
+        if 'exp' in claims and (int(claims['exp']) + TOKEN_EXPIRATION_LEEWAY_SEC) < now:
+            logger.warning('Expired token, exp=%s, now=%s', claims['exp'], now)
+            return None
+
+        if 'nbf' in claims and (int(claims['nbf']) - TOKEN_NOT_BEFORE_LEEWAY_SEC) > now:
+            logger.warning('Early token, nbf=%s, now=%s', claims['nbf'], now)
+            return None
+
+        if 'jti' in claims and not utils.verify_and_burn_nonce(claims['jti']):
+            logger.warning('Invalid nonce: %s', claims['jti'])
+            return None
+
+        return claims
```

Instead of returning `None`, it should raise the exception or return the claims.